cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1211,24 +1211,59 @@ struct get_page_schema {
   __device__ size_type operator()(gpu::PageInfo const& page) { return page.src_col_schema; }
 };
 
+struct input_col_info {
+  int schema_idx;
+  size_t nesting_depth;
+};
+
+/**
+ * @brief Converts a 1-dimensional index into page, depth and column indices used in
+ * allocate_columns to compute columns sizes. The input index will iterate through pages, nesting depth and
+ * column indices in that order.
+ */
+struct get_indices {
+  size_t page_idx;
+  size_t depth_idx;
+  size_t col_idx;
```

Comment:
`col_idx` should be `size_type`, given that `table::num_columns()` return this type. Since we can't have deeper nesting than the number of columns, IMO it's safe to use `size_type` for `depth_idx` as well.
`page_idx` is computed as `index % num_pages`, where index is `int`. So either `index` should be `size_t`, or we should use `int` for `page_idx`. This one only depends on Parquet specs, so I'm not sure what the real limit is.

Reply:
https://github.com/rapidsai/cudf/blob/69abff73b81b05fe2a8d76e07a7ca784f6cf9a3d/cpp/src/io/parquet/reader_impl_helpers.cpp#L401-L403

AFAIK, this is where schema index originate as `size_type`. Originally it was `size_t` stored in `SchemaElement::children_idx`, it's casted to size_type above.

https://github.com/rapidsai/cudf/blob/0af0c5910fa231f6071853ceef3b73827323538a/cpp/src/io/parquet/parquet.hpp#L156